"GET /d HTTP/1.1" 500 59
"GET /d HTTP/1.1" 500 59
"GET /d HTTP/1.1" 500 59
"GET /d HTTP/1.1" 500 59
Internal Server Error: /jobs/create/
Traceback (most recent call last):
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\jobs\views.py", line 24, in create_job
    custom_permission_denied()
    ^^^^^^^^^^^^^^^^^^^^^^^^
NameError: name 'custom_permission_denied' is not defined
"GET /jobs/create/ HTTP/1.1" 500 1034
Internal Server Error: /jobs/create/
Traceback (most recent call last):
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\jobs\views.py", line 24, in create_job
    custom_permission_denied()
    ^^^^^^^^^^^^^^^^^^^^^^^^
NameError: name 'custom_permission_denied' is not defined
"GET /jobs/create/ HTTP/1.1" 500 1034
Internal Server Error: /
Traceback (most recent call last):
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\src\views.py", line 15, in index
    return render(request, 'index.html', {'jobs': jobs, 'is_teacher': is_teacher})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'home' not found. 'home' is not a valid view function or pattern name.
"GET / HTTP/1.1" 500 1034
Internal Server Error: /
Traceback (most recent call last):
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\src\views.py", line 15, in index
    return render(request, 'index.html', {'jobs': jobs, 'is_teacher': is_teacher})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\programming\python\DjangoPr2\pp\.venv\lib\python3.11\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'home' not found. 'home' is not a valid view function or pattern name.
"GET / HTTP/1.1" 500 1034
"GET /r HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
"GET / HTTP/1.1" 500 59
"GET /favicon.ico HTTP/1.1" 500 59
